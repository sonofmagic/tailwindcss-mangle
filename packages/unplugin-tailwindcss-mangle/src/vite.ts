import { unplugin } from './core'
import type {} from 'vite'

export default unplugin.vite
