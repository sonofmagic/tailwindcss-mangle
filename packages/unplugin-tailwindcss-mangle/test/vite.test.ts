import path from 'node:path'
import { build } from 'vite'
import type { RollupOutput } from 'rollup'
import utwm from '@/vite'

const appRoot = path.resolve(__dirname, 'fixtures/vite-repo')
describe('vite build', () => {
  it('common build ', async () => {
    const res = (await build({
      root: appRoot,
      build: {
        write: false,
        cssMinify: false,
        rollupOptions: {
          output: {
            entryFileNames: `[name].js`,
            chunkFileNames: `[name].js`,
            assetFileNames: `[name].[ext]`,
          },
        },
      },
      plugins: [
        utwm({
          classListPath: path.resolve(appRoot, '.tw-patch/tw-class-list.json'),
        }),
      ],
    })) as RollupOutput
    const output = res.output
    expect(output.length).toBe(3)
    const jsFile = output[0]
    expect(jsFile.type).toBe('chunk')
    expect(jsFile.code).toMatchSnapshot()
    expect(jsFile.code).toContain('ease-out')
    expect(output[1].type).toBe('asset')
    if (output[1].type === 'asset') {
      expect(output[1].source).toMatchSnapshot()
    }
    expect(output[2].type).toBe('asset')
    if (output[2].type === 'asset') {
      expect(output[2].source).toMatchSnapshot()
    }
  })

  it('common build change class prefix', async () => {
    const res = (await build({
      root: appRoot,
      build: {
        write: false,
        cssMinify: false,
        rollupOptions: {
          output: {
            entryFileNames: `[name].js`,
            chunkFileNames: `[name].js`,
            assetFileNames: `[name].[ext]`,
          },
        },
      },
      plugins: [
        utwm({
          classListPath: path.resolve(appRoot, '.tw-patch/tw-class-list.json'),
          classGenerator: {
            classPrefix: 'ice-',
          },
        }),
      ],
    })) as RollupOutput
    const output = res.output
    expect(output.length).toBe(3)
    const jsFile = output[0]
    expect(jsFile.type).toBe('chunk')
    expect(jsFile.code).toMatchSnapshot()
    expect(jsFile.code).toContain('ease-out')
    expect(output[1].type).toBe('asset')
    if (output[1].type === 'asset') {
      expect(output[1].source).toMatchSnapshot()
    }
    expect(output[2].type).toBe('asset')
    if (output[2].type === 'asset') {
      expect(output[2].source).toMatchSnapshot()
    }
  })
})
