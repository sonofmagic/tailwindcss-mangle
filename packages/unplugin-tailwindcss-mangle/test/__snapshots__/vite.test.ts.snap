// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vite build > common build  1`] = `
"(function(){const a=document.createElement(\\"link\\").relList;if(a&&a.supports&&a.supports(\\"modulepreload\\"))return;for(const t of document.querySelectorAll('link[rel=\\"modulepreload\\"]'))c(t);new MutationObserver(t=>{for(const e of t)if(e.type===\\"childList\\")for(const w of e.addedNodes)w.tagName===\\"LINK\\"&&w.rel===\\"modulepreload\\"&&c(w)}).observe(document,{childList:!0,subtree:!0});function s(t){const e={};return t.integrity&&(e.integrity=t.integrity),t.referrerPolicy&&(e.referrerPolicy=t.referrerPolicy),t.crossOrigin===\\"use-credentials\\"?e.credentials=\\"include\\":t.crossOrigin===\\"anonymous\\"?e.credentials=\\"omit\\":e.credentials=\\"same-origin\\",e}function c(t){if(t.ep)return;t.ep=!0;const e=s(t);fetch(t.href,e)}})();function n(r){r.animate({clipPath:[]},{duration:400,easing:\\"ease-out\\"})}document.querySelector(\\"#app\\").innerHTML=\`
  <div>
    <button id=\\"counter\\">counter</button>
  </div>

<main class=\\"flex tw-fb tw-yb tw-cb tw-ra tw-mc\\">
<div class=\\"tw-rc tw-gc tw-ub tw-cb tw-ra tw-ob tw-cc tw-ac\\">
  <p
    class=\\"fixed tw-dc tw-jc flex tw-gc tw-va tw-wb tw-pa tw-ja tw-za tw-nc tw-oc tw-ca tw-l tw-v tw-ea tw-sb tw-tb  tw-ab tw-qb tw-wa tw-fc tw-m\\">
    Get started by editing&nbsp;
    <code class=\\"tw-ob tw-nb\\">pages/index.tsx</code>
  </p>
  <div
    class=\\"fixed tw-xb tw-dc flex tw-kc tw-gc tw-pb tw-va tw-ka tw-jb tw-vb tw-qa tw-ua tw-sb tw-rb tw-tb tw-kb\\">
    <a class=\\"tw-w flex tw-ba tw-hc tw-tc tw-n tw-ec\\"
      href=\\"https://vercel.com?utm_source=create-next-app&utm_medium=default-template-tw&utm_campaign=create-next-app\\"
      target=\\"_blank\\" rel=\\"noopener noreferrer\\">
      By
    </a>
  </div>
</div>

<div
  class=\\"relative flex tw-ba tw-na tw-ha tw-ia tw-k tw-u before:bg-gradient-radial tw-da tw-p tw-oa tw-s tw-sa tw-gb tw-la tw-ma tw-o after:bg-gradient-conic tw-y tw-z tw-ta tw-x tw-b tw-d tw-h tw-j tw-g tw-t\\">
  <img class=\\"relative tw-a tw-hb\\" src=\\"/vite.svg\\" alt=\\"Next.js Logo\\"
    priority />
</div>

<div class=\\"tw-ic grid tw-ib tw-bc tw-xa tw-db\\">
  <a href=\\"https://nextjs.org/docs?utm_source=create-next-app&utm_medium=default-template-tw&utm_campaign=create-next-app\\"
    class=\\"group tw-mb border tw-aa tw-pc tw-qc tw-ga tw-q tw-fa tw-c tw-e\\"
    target=\\"_blank\\" rel=\\"noopener noreferrer\\">
    <h2 class=\\"tw-lc tw-zb tw-ya\\">
      Docs <span
        class=\\"tw-bb tw-r tw-i tw-f\\">-&gt;</span>
    </h2>
    <p class=\\"tw-sc tw-eb tw-cc tw-lb\\">Find in-depth information about Next.js
      features and API.</p>
  </a>

  <a href=\\"https://nextjs.org/learn?utm_source=create-next-app&utm_medium=default-template-tw&utm_campaign=create-next-app\\"
    class=\\"group tw-mb border tw-aa tw-pc tw-qc tw-ga tw-q tw-fa tw-c tw-e\\"
    target=\\"_blank\\" rel=\\"noopener noreferrer\\">
    <h2 class=\\"tw-lc tw-zb tw-ya\\">
      Learn <span
        class=\\"tw-bb tw-r tw-i tw-f\\">-&gt;</span>
    </h2>
    <p class=\\"tw-sc tw-eb tw-cc tw-lb\\">Learn about Next.js in an interactive
      course with&nbsp;quizzes!</p>
  </a>

  <a href=\\"https://vercel.com/templates?framework=next.js&utm_source=create-next-app&utm_medium=default-template-tw&utm_campaign=create-next-app\\"
    class=\\"group tw-mb border tw-aa tw-pc tw-qc tw-ga tw-q tw-fa tw-c tw-e\\"
    target=\\"_blank\\" rel=\\"noopener noreferrer\\">
    <h2 class=\\"tw-lc tw-zb tw-ya\\">
      Templates <span
        class=\\"tw-bb tw-r tw-i tw-f\\">-&gt;</span>
    </h2>
    <p class=\\"tw-sc tw-eb tw-cc tw-lb\\">Discover and deploy boilerplate example
      Next.js&nbsp;projects.</p>
  </a>

  <a href=\\"https://vercel.com/new?utm_source=create-next-app&utm_medium=default-template-tw&utm_campaign=create-next-app\\"
    class=\\"group tw-mb border tw-aa tw-pc tw-qc tw-ga tw-q tw-fa tw-c tw-e\\"
    target=\\"_blank\\" rel=\\"noopener noreferrer\\">
    <h2 class=\\"tw-lc tw-zb tw-ya\\">
      Deploy <span
        class=\\"tw-bb tw-r tw-i tw-f\\">-&gt;</span>
    </h2>
    <p class=\\"tw-sc tw-eb tw-cc tw-lb\\">Instantly deploy your Next.js site to a
      shareable URL with Vercel.</p>
  </a>
</div>
</main>
\`;n(document.querySelector(\\"#counter\\"));
"
`;

exports[`vite build > common build  2`] = `
"*,::before,::after {
    --tw-border-spacing-x: 0;
    --tw-border-spacing-y: 0;
    --tw-translate-x: 0;
    --tw-translate-y: 0;
    --tw-rotate: 0;
    --tw-skew-x: 0;
    --tw-skew-y: 0;
    --tw-scale-x: 1;
    --tw-scale-y: 1;
    --tw-pan-x:  ;
    --tw-pan-y:  ;
    --tw-pinch-zoom:  ;
    --tw-scroll-snap-strictness: proximity;
    --tw-gradient-from-position:  ;
    --tw-gradient-via-position:  ;
    --tw-gradient-to-position:  ;
    --tw-ordinal:  ;
    --tw-slashed-zero:  ;
    --tw-numeric-figure:  ;
    --tw-numeric-spacing:  ;
    --tw-numeric-fraction:  ;
    --tw-ring-inset:  ;
    --tw-ring-offset-width: 0px;
    --tw-ring-offset-color: #fff;
    --tw-ring-color: rgb(59 130 246 / 0.5);
    --tw-ring-offset-shadow: 0 0 #0000;
    --tw-ring-shadow: 0 0 #0000;
    --tw-shadow: 0 0 #0000;
    --tw-shadow-colored: 0 0 #0000;
    --tw-blur:  ;
    --tw-brightness:  ;
    --tw-contrast:  ;
    --tw-grayscale:  ;
    --tw-hue-rotate:  ;
    --tw-invert:  ;
    --tw-saturate:  ;
    --tw-sepia:  ;
    --tw-drop-shadow:  ;
    --tw-backdrop-blur:  ;
    --tw-backdrop-brightness:  ;
    --tw-backdrop-contrast:  ;
    --tw-backdrop-grayscale:  ;
    --tw-backdrop-hue-rotate:  ;
    --tw-backdrop-invert:  ;
    --tw-backdrop-opacity:  ;
    --tw-backdrop-saturate:  ;
    --tw-backdrop-sepia:  
}
::backdrop {
    --tw-border-spacing-x: 0;
    --tw-border-spacing-y: 0;
    --tw-translate-x: 0;
    --tw-translate-y: 0;
    --tw-rotate: 0;
    --tw-skew-x: 0;
    --tw-skew-y: 0;
    --tw-scale-x: 1;
    --tw-scale-y: 1;
    --tw-pan-x:  ;
    --tw-pan-y:  ;
    --tw-pinch-zoom:  ;
    --tw-scroll-snap-strictness: proximity;
    --tw-gradient-from-position:  ;
    --tw-gradient-via-position:  ;
    --tw-gradient-to-position:  ;
    --tw-ordinal:  ;
    --tw-slashed-zero:  ;
    --tw-numeric-figure:  ;
    --tw-numeric-spacing:  ;
    --tw-numeric-fraction:  ;
    --tw-ring-inset:  ;
    --tw-ring-offset-width: 0px;
    --tw-ring-offset-color: #fff;
    --tw-ring-color: rgb(59 130 246 / 0.5);
    --tw-ring-offset-shadow: 0 0 #0000;
    --tw-ring-shadow: 0 0 #0000;
    --tw-shadow: 0 0 #0000;
    --tw-shadow-colored: 0 0 #0000;
    --tw-blur:  ;
    --tw-brightness:  ;
    --tw-contrast:  ;
    --tw-grayscale:  ;
    --tw-hue-rotate:  ;
    --tw-invert:  ;
    --tw-saturate:  ;
    --tw-sepia:  ;
    --tw-drop-shadow:  ;
    --tw-backdrop-blur:  ;
    --tw-backdrop-brightness:  ;
    --tw-backdrop-contrast:  ;
    --tw-backdrop-grayscale:  ;
    --tw-backdrop-hue-rotate:  ;
    --tw-backdrop-invert:  ;
    --tw-backdrop-opacity:  ;
    --tw-backdrop-saturate:  ;
    --tw-backdrop-sepia:  
}
.tw-w {
    pointer-events: none
}
.fixed {
    position: fixed
}
.relative {
    position: relative
}
.tw-xb {
    bottom: 0px
}
.tw-dc {
    left: 0px
}
.tw-jc {
    top: 0px
}
.tw-rc {
    z-index: 10
}
.tw-sc {
    margin: 0px
}
.tw-lc {
    margin-bottom: 0.75rem
}
.tw-ic {
    margin-bottom: 8rem
}
.tw-bb {
    display: inline-block
}
.flex {
    display: flex
}
.grid {
    display: grid
}
.tw-kc {
    height: 12rem
}
.tw-fb {
    min-height: 100vh
}
.tw-gc {
    width: 100%
}
.tw-ub {
    max-width: 64rem
}
.tw-eb {
    max-width: 30ch
}
.tw-yb {
    flex-direction: column
}
.tw-ba {
    place-items: center
}
.tw-pb {
    align-items: flex-end
}
.tw-cb {
    align-items: center
}
.tw-va {
    justify-content: center
}
.tw-ra {
    justify-content: space-between
}
.tw-hc {
    gap: 0.5rem
}
.tw-mb {
    border-radius: 0.5rem
}
.border {
    border-width: 1px
}
.tw-wb {
    border-bottom-width: 1px
}
.tw-pa {
    --tw-border-opacity: 1;
    border-color: rgb(209 213 219 / var(--tw-border-opacity))
}
.tw-aa {
    border-color: transparent
}
.tw-ja {
    background-image: linear-gradient(to bottom, var(--tw-gradient-stops))
}
.tw-ka {
    background-image: linear-gradient(to top, var(--tw-gradient-stops))
}
.tw-jb {
    --tw-gradient-from: #fff var(--tw-gradient-from-position);
    --tw-gradient-to: rgb(255 255 255 / 0) var(--tw-gradient-to-position);
    --tw-gradient-stops: var(--tw-gradient-from), var(--tw-gradient-to)
}
.tw-za {
    --tw-gradient-from: #e4e4e7 var(--tw-gradient-from-position);
    --tw-gradient-to: rgb(228 228 231 / 0) var(--tw-gradient-to-position);
    --tw-gradient-stops: var(--tw-gradient-from), var(--tw-gradient-to)
}
.tw-vb {
    --tw-gradient-to: rgb(255 255 255 / 0)  var(--tw-gradient-to-position);
    --tw-gradient-stops: var(--tw-gradient-from), #fff var(--tw-gradient-via-position), var(--tw-gradient-to)
}
.tw-mc {
    padding: 6rem
}
.tw-tc {
    padding: 2rem
}
.tw-pc {
    padding-left: 1.25rem;
    padding-right: 1.25rem
}
.tw-qc {
    padding-top: 1rem;
    padding-bottom: 1rem
}
.tw-nc {
    padding-bottom: 1.5rem
}
.tw-oc {
    padding-top: 2rem
}
.tw-ib {
    text-align: center
}
.tw-ob {
    font-family: ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, \\"Liberation Mono\\", \\"Courier New\\", monospace
}
.tw-zb {
    font-size: 1.5rem;
    line-height: 2rem
}
.tw-cc {
    font-size: 0.875rem;
    line-height: 1.25rem
}
.tw-nb {
    font-weight: 700
}
.tw-ya {
    font-weight: 600
}
.tw-lb {
    opacity: 0.5
}
.tw-ca {
    --tw-backdrop-blur: blur(40px);
    backdrop-filter: var(--tw-backdrop-blur) var(--tw-backdrop-brightness) var(--tw-backdrop-contrast) var(--tw-backdrop-grayscale) var(--tw-backdrop-hue-rotate) var(--tw-backdrop-invert) var(--tw-backdrop-opacity) var(--tw-backdrop-saturate) var(--tw-backdrop-sepia)
}
.tw-ga {
    transition-property: color, background-color, border-color, text-decoration-color, fill, stroke;
    transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
    transition-duration: 150ms
}
.tw-r {
    transition-property: transform;
    transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
    transition-duration: 150ms
}
.ease-out {
    transition-timing-function: cubic-bezier(0, 0, 0.2, 1)
}
.tw-na::before {
    content: var(--tw-content);
    position: absolute
}
.tw-ha::before {
    content: var(--tw-content);
    height: 300px
}
.tw-ia::before {
    content: var(--tw-content);
    width: 480px
}
.tw-k::before {
    content: var(--tw-content);
    --tw-translate-x: -50%;
    transform: translate(var(--tw-translate-x), var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))
}
.tw-u::before {
    content: var(--tw-content);
    border-radius: 9999px
}
.tw-da::before {
    content: var(--tw-content);
    --tw-gradient-from: #fff var(--tw-gradient-from-position);
    --tw-gradient-to: rgb(255 255 255 / 0) var(--tw-gradient-to-position);
    --tw-gradient-stops: var(--tw-gradient-from), var(--tw-gradient-to)
}
.tw-p::before {
    content: var(--tw-content);
    --tw-gradient-to: transparent var(--tw-gradient-to-position)
}
.tw-oa::before {
    content: var(--tw-content);
    --tw-blur: blur(40px);
    filter: var(--tw-blur) var(--tw-brightness) var(--tw-contrast) var(--tw-grayscale) var(--tw-hue-rotate) var(--tw-invert) var(--tw-saturate) var(--tw-sepia) var(--tw-drop-shadow)
}
.tw-s::before {
    --tw-content: '';
    content: var(--tw-content)
}
.tw-sa::after {
    content: var(--tw-content);
    position: absolute
}
.tw-gb::after {
    content: var(--tw-content);
    z-index: -20
}
.tw-la::after {
    content: var(--tw-content);
    height: 180px
}
.tw-ma::after {
    content: var(--tw-content);
    width: 240px
}
.tw-o::after {
    content: var(--tw-content);
    --tw-translate-x: 33.333333%;
    transform: translate(var(--tw-translate-x), var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))
}
.tw-y::after {
    content: var(--tw-content);
    --tw-gradient-from: #bae6fd var(--tw-gradient-from-position);
    --tw-gradient-to: rgb(186 230 253 / 0) var(--tw-gradient-to-position);
    --tw-gradient-stops: var(--tw-gradient-from), var(--tw-gradient-to)
}
.tw-z::after {
    content: var(--tw-content);
    --tw-gradient-to: rgb(191 219 254 / 0)  var(--tw-gradient-to-position);
    --tw-gradient-stops: var(--tw-gradient-from), #bfdbfe var(--tw-gradient-via-position), var(--tw-gradient-to)
}
.tw-ta::after {
    content: var(--tw-content);
    --tw-blur: blur(40px);
    filter: var(--tw-blur) var(--tw-brightness) var(--tw-contrast) var(--tw-grayscale) var(--tw-hue-rotate) var(--tw-invert) var(--tw-saturate) var(--tw-sepia) var(--tw-drop-shadow)
}
.tw-x::after {
    --tw-content: '';
    content: var(--tw-content)
}
.tw-q:hover {
    --tw-border-opacity: 1;
    border-color: rgb(209 213 219 / var(--tw-border-opacity))
}
.tw-fa:hover {
    --tw-bg-opacity: 1;
    background-color: rgb(243 244 246 / var(--tw-bg-opacity))
}
.group:hover .tw-i {
    --tw-translate-x: 0.25rem;
    transform: translate(var(--tw-translate-x), var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))
}
@media (prefers-reduced-motion: reduce) {
    .tw-f {
        transform: none
    }
}
@media (prefers-color-scheme: dark) {
    .tw-l {
        --tw-border-opacity: 1;
        border-color: rgb(38 38 38 / var(--tw-border-opacity))
    }
    .tw-v {
        background-color: rgb(39 39 42 / 0.3)
    }
    .tw-qa {
        --tw-gradient-from: #000 var(--tw-gradient-from-position);
        --tw-gradient-to: rgb(0 0 0 / 0) var(--tw-gradient-to-position);
        --tw-gradient-stops: var(--tw-gradient-from), var(--tw-gradient-to)
    }
    .tw-ea {
        --tw-gradient-from: inherit var(--tw-gradient-from-position);
        --tw-gradient-to: rgb(255 255 255 / 0) var(--tw-gradient-to-position);
        --tw-gradient-stops: var(--tw-gradient-from), var(--tw-gradient-to)
    }
    .tw-ua {
        --tw-gradient-to: rgb(0 0 0 / 0)  var(--tw-gradient-to-position);
        --tw-gradient-stops: var(--tw-gradient-from), #000 var(--tw-gradient-via-position), var(--tw-gradient-to)
    }
    .tw-a {
        --tw-drop-shadow: drop-shadow(0 0 0.3rem #ffffff70);
        filter: var(--tw-blur) var(--tw-brightness) var(--tw-contrast) var(--tw-grayscale) var(--tw-hue-rotate) var(--tw-invert) var(--tw-saturate) var(--tw-sepia) var(--tw-drop-shadow)
    }
    .tw-hb {
        --tw-invert: invert(100%);
        filter: var(--tw-blur) var(--tw-brightness) var(--tw-contrast) var(--tw-grayscale) var(--tw-hue-rotate) var(--tw-invert) var(--tw-saturate) var(--tw-sepia) var(--tw-drop-shadow)
    }
    .tw-b::before {
        content: var(--tw-content);
        background-image: linear-gradient(to bottom right, var(--tw-gradient-stops))
    }
    .tw-d::before {
        content: var(--tw-content);
        --tw-gradient-from: transparent var(--tw-gradient-from-position);
        --tw-gradient-to: rgb(0 0 0 / 0) var(--tw-gradient-to-position);
        --tw-gradient-stops: var(--tw-gradient-from), var(--tw-gradient-to)
    }
    .tw-h::before {
        content: var(--tw-content);
        --tw-gradient-to: rgb(29 78 216 / 0.1) var(--tw-gradient-to-position)
    }
    .tw-j::after {
        content: var(--tw-content);
        --tw-gradient-from: #0c4a6e var(--tw-gradient-from-position);
        --tw-gradient-to: rgb(12 74 110 / 0) var(--tw-gradient-to-position);
        --tw-gradient-stops: var(--tw-gradient-from), var(--tw-gradient-to)
    }
    .tw-g::after {
        content: var(--tw-content);
        --tw-gradient-to: rgb(1 65 255 / 0)  var(--tw-gradient-to-position);
        --tw-gradient-stops: var(--tw-gradient-from), rgb(1 65 255 / 0.4) var(--tw-gradient-via-position), var(--tw-gradient-to)
    }
    .tw-c:hover {
        --tw-border-opacity: 1;
        border-color: rgb(64 64 64 / var(--tw-border-opacity))
    }
    .tw-e:hover {
        background-color: rgb(38 38 38 / 0.3)
    }
}
@media (min-width: 1024px) {
    .tw-n {
        pointer-events: auto
    }
    .tw-sb {
        position: static
    }
    .tw-bc {
        margin-bottom: 0px
    }
    .tw-ac {
        display: flex
    }
    .tw-rb {
        height: auto
    }
    .tw-tb {
        width: auto
    }
    .tw-xa {
        grid-template-columns: repeat(4, minmax(0, 1fr))
    }
    .tw-ab {
        border-radius: 0.75rem
    }
    .tw-qb {
        border-width: 1px
    }
    .tw-wa {
        --tw-bg-opacity: 1;
        background-color: rgb(229 231 235 / var(--tw-bg-opacity))
    }
    .tw-kb {
        background-image: none
    }
    .tw-ec {
        padding: 0px
    }
    .tw-fc {
        padding: 1rem
    }
    .tw-db {
        text-align: left
    }
    .tw-t::before {
        content: var(--tw-content);
        height: 360px
    }
    @media (prefers-color-scheme: dark) {
        .tw-m {
            background-color: rgb(39 39 42 / 0.3)
        }
    }
}
"
`;

exports[`vite build > common build  3`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\" />
    <link rel=\\"icon\\" type=\\"image/svg+xml\\" href=\\"/vite.svg\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <title>Vite + TS</title>
    <script type=\\"module\\" crossorigin src=\\"/index.js\\"></script>
    <link rel=\\"stylesheet\\" href=\\"/index.css\\">
  </head>
  <body>
    <div id=\\"app\\"></div>
    
  </body>
</html>
"
`;
